examples/onboard-azure-ms-sql-server/main.tf

```diff
@@ -38,94 +37,84 @@ provider "dsfhub" {
-  resource_group_name = local.azure_resource_group_name
-  send                = true
+  storage_account_location            = local.azure_location
+  storage_account_name                = "sqlserverstorageacc"
+  storage_account_resource_group_name = local.azure_resource_group_name
+  storage_container_name              = "sqlserverstoragecon"
 }
 
-module "eventhub-read-authorization" {
-  source = "../../modules/azurerm-eventhub-namespace-authorization-rule"
-
-  listen              = true
-  manage              = false
-  name                = "${local.azure_eventhub_name}read"
-  namespace_name      = module.eventhub-namespace.this.name
-  resource_group_name = local.azure_resource_group_name
-  send                = false
-}
+################################################################################
+# Azure SQL Server
+################################################################################
+module "azure-ms-sql-server-1" {
+  source = "../../modules/onboard-azure-ms-sql-server"
 
-# 2. Storage Account and Container
-module "storage-account" {
-  source = "../../modules/azurerm-storage-account"
+  depends_on = [module.onboard-azure-sql-server-eventhub-1]
 
-  location            = local.azure_location
-  name                = "sqlserverstorageacc"
-  resource_group_name = local.azure_resource_group_name
-}
+  azure_ms_sql_server_admin_email               = local.admin_email
+  azure_ms_sql_server_audit_pull_enabled        = true
+  azure_ms_sql_server_gateway_id                = local.gateway_id
+  azure_ms_sql_server_location                  = local.azure_location
+  azure_ms_sql_server_logs_destination_asset_id = module.onboard-azure-sql-server-eventhub-1.azure-eventhub-asset.asset_id
 
-module "storage-container" {
-  source = "../../modules/azurerm-storage-container"
+  diagnostic_setting_eventhub_authorization_rule_id = module.onboard-azure-sql-server-eventhub-1.eventhub-write-authorization.id
+  diagnostic_setting_eventhub_name                  = module.onboard-azure-sql-server-eventhub-1.eventhub.name
+  diagnostic_setting_name                           = "dsfhubdiagnostic"
 
-  name                 = "sqlserverstoragecon"
-  storage_account_name = module.storage-account.this.name
+  server_administrator_login           = "exampleadmin"
+  server_administrator_login_password  = "Abcd1234"
+  server_location                      = local.azure_location
+  server_name                          = "example-azure-sql-server"
+  server_public_network_access_enabled = true
+  server_resource_group_name           = local.azure_resource_group_name
 }
 
 ################################################################################
-# Azure SQL Server
+# Azure SQL Server Many-to-One
 ################################################################################
-module "azure-ms-sql-server-1" {
+locals {
+  sql_server_types = toset([
+    "dev",
+    "prod",
+    "uat"
+  ])
+}
```

Comment:
Just wanted to say that I'm a big fan of this lil refactor and I like the example you picked here too! 💡

Reply:
I think it's a lot cleaner as well! 

As for the example, I hear clients say it enough, may as well use em!